pages/reference/OS/configuration.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Configuration
+excerpt: Instructions for editing and customizing config.json for modifying the behavior of {{ $names.os.lower }}
+---
+
+# Configuring {{ $names.os.lower }}
+
+The behavior of {{ $names.os.lower }} can be configured by editing the `config.json` file. This file is located in the [boot partition][boot-partition] and accepts a [range of fields](#valid-fields) to modify the behavior of the host OS.
+
+Before the device has been provisioned, you may edit `config.json` by mounting a flashed SD card (which will mount as `resin-boot`) and editing the file directly. The boot partition is mounted on the device at `/mnt/boot`, and so on the device, the file is located at `/mnt/boot/config.json`. For example, to output a formatted version of `config.json` on a device, use the following commands:
```

Comment:
* has been provisioned → is provisioned
* This paragraph describes how to edit the `config.json` file ***before*** the device is provisioned. Then there is a whole section **"Sample config.json",** a long section **"Valid fields",** and then the section **"Editing config.json on a provisioned device".** I think this page would be better organised by having the complementary options of editing `config.json` _before and after_ provisioning in the same section, or in consecutive sections, rather than in "distant" sections.
* I am not sure how obvious the meaning of "provisioned" is. I suggest making the word "provisioned" a link to https://www.balena.io/docs/learn/welcome/primer/#device-provisioning

Reply:
Thanks again for the feedback. The distinction between before and after provisioning is _really_ confusing as the behavior is inconsistent. I did a lot of testing on this and determined the following behavior:

* Before or after provisioning you can edit _some_ values of config.json so long as the Supervisor doesn't manage them. I mounted an SD card from a provisioned device into my laptop and added the SSH keys or udev rules, for example, and that works just fine.
* When you edit `persistentLogging` before the device is provisioned this would be respected but if you edit this _after_ the device has been provisioned the Supervisor overwrites this setting - it actually reformats the file too (to minify the json). I suspect the same would be true of the hostname too (I'll test that later)

So the distinction between before and after provisioning is very muddled and has caveats, but I may just make this a single section and mention these.

Reply:
@garethtdavies, what you've described above matches my previous "anecdotal" experience -- but I had not conducted an experiment (or read source code...) to allow me to state it. It really sounds correct though, and actually perhaps you could add a sentence in the lines of:
_"After provisioning, editing the `config.json` as described above is not reliable or advisable because the supervisor may overwrite some fields, such as persistentLogging, with values read from the balenaCloud (API) database."_
Just a thought. :-)

Reply:
I think we are getting there.... 😆 This behavior is also confirmed by configizer which only allows specifying of fields not managed by the Supervisor.

Thanks for that suggestion. I think something like that will work and am reformating this now. I don't want to blanket advise against modifying it (with a suitable warning) as for example, adding an SSH key for a prod image is incredibly useful. Those changes _do_ work consistently and I could imagine being popular requests.

Reply:
I pushed an update and reorganized so it flows a bit better - i.e. the how to safely edit (three different ways depending on what you want to do...) follows the advice to not do this directly on a provisioned device.

Reply:
Is there a reliable list of which fields can be edited in `config.json` before provisioning (all of them?), and which fields _shouldn't_ be edited after provisioning, because the supervisor will overwrite any changes? Or is that wildly dependent on the particular version of the supervisor?

Reply:
> Is there a reliable list of which fields can be edited in `config.json`

I may be misreading the supervisor source code, but it would appear that any of the fields listed in this [schema.ts](https://github.com/balena-io/balena-supervisor/blob/v10.6.26/src/config/schema.ts) file (like `uuid`, `apiKey`, `hostname`, `persistentLogging`...) are managed by the supervisor, and thus cannot be edited manually. Then this block of code seems to read values from a database and update `config.json`:
https://github.com/balena-io/balena-supervisor/blob/v10.6.26/src/config/index.ts#L153-L180

But I am not 100% sure, and perhaps @CameronDiver could comment further. Also, I don't know if it is a good idea to publish such a list, because this might prevent the supervisor from managing the fields in the future. Unless the list was explicitly linked to supervisor versions, alongside a warning that updating the host OS might break the user's app or scripts if they are "manually" editing some fields...

Reply:
I assumed the list here (valid fields) was the list of editable fields. 

My understanding then is after provisioning hostname (supervisor API) and persistent logging  (config variable) and the rest we recommend configizer (which only allows these other fields). I guess this is why we don't want users to manually edit this file as it can be unpredictable as noted in the text :)